cdci_data_analysis/analysis/parameters.py

```diff
@@ -302,10 +302,18 @@ def set_value_from_form(self, form, verbose=False):
     def set_par(self, value, units=None):
         if units is not None:
             self.units = units
+
         self.value = value
-        return self.get_value_in_default_format()
+        return self.get_value_in_default_units()
+
+    def get_value_in_default_units(self):
+        return self.value
 
+    # to ensure backward compatibility
     def get_value_in_default_format(self):
+        return self.get_value_in_default_units()
+
+    def get_value_in_units(self, units):
```

Comment:
why is this function not using any units?

Reply:
Because it's supposed to be extended by each child class of the class `Parameter`

Reply:
if there is no generic way, this function should raise an exception. Currently it does not do what it implies.

Reply:
Several Parameter(s) do not have an implementation of such a method, thus causing several tests (also in the plugin) to fail.

We can assume to have it returning the very same value in case no implementation is provided.

Reply:
I think it's because only `Float` parameters should have  units. And also all `Float` parameters have to have units.
The fact that the tests fail while this change appears natural means that the logic is a bit unfortunate.
Maybe moving units to `Float` and inheriting those parameters which have units from `Float` would restore the logic.

Reply:
> Maybe moving units to `Float` and inheriting those parameters which have units from `Float` would restore the logic.

You mean to have parameters like `Angle` to inherit from `Float`? But for certain parameters, that cannot really apply, like `Name`.